core/src/main/java/jenkins/model/lazy/BuildReferenceMapAdapter.java

```diff
@@ -392,13 +392,13 @@ public int hashCode() {
         }
 
         private Entry<Integer,BuildReference<R>> _wrap(Entry<Integer,R> e) {
-            return new MapEntry(e.getKey(),wrap(e.getValue()));
+            return new AbstractMap.SimpleEntry<>(e.getKey(),wrap(e.getValue()));
```

Comment:
The only actual change of substance in this PR. `groovy.util.MapEntry` seemed like an odd implementation of a map entry, as it is meant for use by Groovy and just happens to be on the classpath. This is also the only import from `groovy.util` in `src/main/` and seemed like an oversight. The implementation in `AbstractMap.SimpleEntry` is essentially identical, and it seems better to rely on the Java Platform for this than on Groovy. Note that I intentionally did _not_ choose `AbstractMap.SimpleImmutableEntry`, since the Groovy version is mutable and I did not want to change the behavior here.